File: src/models/Projet.ts
import { Carte } from "./Carte";
import { Tableau } from "./Tableau";
import { Utilisateur } from "./Utilisateur";

export class Projet {
    idProjet?: number;
    nom?: string;
    description?: string;
    tableau!: Tableau; // Assuming Tableau interface matches the structure
    cartes!: Carte[];
    utilisateurs!: Utilisateur[]; 
    showFullDescription?: boolean;
    nombreMembres?: number;
    membersList?: any[]; 
    showMembersList?: boolean; 
  }

File: src/main/java/com/example/gestionprojeet/classes/Projet.java
package com.example.gestionprojeet.classes;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Projet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long idProjet   ;
    private String nom     ;
    private  String description;
    @ManyToOne
    private Tableau tableau;
    @OneToMany(mappedBy = "projet")
    private List<Carte> cartes;
    @ManyToMany(mappedBy = "projets")
    private List<Utlisateur> utilisateurs;


}
